osxcollector/osxcollector.py

```diff
@@ -1205,6 +1208,22 @@ def _collect_system_info(self):
         }
         Logger.log_dict(record)
 
+    def _collect_binary_names_in_path(self):
+        """Collect the names of executable binaries in the PATH environment"""
+        exe_files = []
+
+        def is_exe(fpath):
+            return os.path.isfile(fpath) and os.path.exists(fpath) and os.access(fpath, os.X_OK)
```

Comment:
`os.path.isfile` always implies `os.path.exists`, so the latter seems redundant. You could also have a directory mislabelled as an executable which would probably further encourage just using `os.path.isfile` and `os.access` alone.

Reply:
ditto